domain/src/test/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/StudentUnpreparedSteps.kt

```diff
@@ -0,0 +1,136 @@
+
+class StudentUnpreparedSteps : En {
+
+    lateinit var attendance: CheckedAttendanceList
+    lateinit var lessonIdentifier: LessonIdentifier
+    lateinit var currentLesson: CurrentLesson
+    lateinit var studentsUnpreparedForLesson: StudentsUnpreparedForLesson
+    lateinit var studentSubjectUnpreparednessInASemester: StudentSubjectUnpreparednessInASemester
+
+    val reportUnpreparednessFacade: ReportUnpreparedness = reportUnpreparedness(
+        checkStudentCanReportUnprepared(checkNumberOfTimesStudentWasUnpreparedInSemester {
+            studentSubjectUnpreparednessInASemester
+                .let { AsyncFactory.justSuccess(it) }
+        }, hasStudentUsedAllUnpreparednesses),
+        noteStudentUnprepared(hasStudentAlreadyRaisedUnprepared),
+        checkStudentIsPresent(areStudentsEqual),
+        createEvent
+    )
+
+    lateinit var result: Output<StudentMarkedUnprepared, UnpreparednessError>
+
+    init {
+        Given("Present {word} {word} from class {word}") { firstName: String, secondName: String, className: String ->
+            attendance = CheckedAttendanceList(
+                listOf(PresentStudent(
+                    FirstName(NonEmptyText.of(firstName)!!),
+                    SecondName(NonEmptyText.of(secondName)!!),
+                    NumberInRegister(NaturalNumber.ONE))
+                ),
+                emptyList()
+            )
+            lessonIdentifier = LessonIdentifier(
+                LocalDate.now(),
+                LessonHourNumber.of(1).orNull()!!,
+                ClassName(NonEmptyText.of(className)!!)
+            )
+        }
+        Given("Lesson after attendance checked but before topic assigned") {
+            currentLesson = LessonAfterAttendance(lessonIdentifier, attendance, studentsUnpreparedForLesson)
+        }
+        Given("Lesson after topic assigned") {
+            val anyTopic = LessonTopic(
+                LessonOrdinalNumber(NaturalNumber.ONE),
+                TopicTitle(NonEmptyText.of("Wingardium Leviosa charm")!!),
+                LocalDate.now()
+            )
+            currentLesson = InProgressLesson(lessonIdentifier, attendance, studentsUnpreparedForLesson, anyTopic)
+        }
+        Given("{word} {word} reported unprepared {int} times in a semester") { firstName: String, secondName: String, number: Int ->
+            val student = StudentInSemester(
+                lessonIdentifier.className,
+                FirstName(NonEmptyText.of(firstName)!!),
+                SecondName(NonEmptyText.of(secondName)!!)
+            )
+
+            studentSubjectUnpreparednessInASemester =
+                StudentSubjectUnpreparednessInASemester.create(number, student).orNull()!!
+        }
+        Given("Empty list of unprepared students") {
+            studentsUnpreparedForLesson = StudentsUnpreparedForLesson()
+        }
+        Given("{word} {word} on the list of unprepared students") { firstName: String, secondName: String ->
```

Comment:
I understand that it's kind of limitation of Cucumber, but multiple `Given\Then` sections puzzle me... :man_shrugging:

Reply:
what would you like it to look like?

Reply:
separate Classes/Functions which encapsulates singular scenario from Gherkin